flow/layers/layer.h

```diff
@@ -66,6 +66,11 @@ struct PrerollContext {
   float total_elevation = 0.0f;
   bool has_platform_view = false;
   bool is_opaque = true;
+#if defined(OS_FUCHSIA)
+  // True if, during the traversal so far, we have seen a child_scene_layer.
+  // Informs whether a layer needs to be system composited.
+  bool child_scene_layer_exists_below = false;
```

Comment:
Why not use needs_system_composite() for this?  That is the intent of the flag, and with your other changes the *only* place it is set is inside of child_scene_layer.  It would be perfect for the job and you could avoid adding all of the logic related to this flag

Reply:
needs_system_composite() is for a particular layer, while this is defined for the *PrerollContext*. we need to track this value during a traversal, so physicalshapelayers can decide whether or not they should be system composited-- a big part of this change is that they usually are *not* system composited.

Reply:
Yes, while needs_system_composite() is for a particular layer, ContainerLayer::PrerollChildren ensures that the needs_system_composite flags is propagated up the tree (or forward in the traversal order, if you like).

The means that every single PhysicalShapeLayer renderered before a ChildSceneLayer in the traversal order will already have the needs_system_composite() flag applied to it.

Reply:
FWIW, I actually prefer your method of storing this in the context after thinking about this more....it prevents carrying a (mostly) useless bool inside of every single Layer.

I think you could also accomplish this by relying on your flag completely and deleting needs_system_composite().  The 2 flags are IMO redundant with one another, so one of them should go.

Reply:
I think you're right that it's propagated up the tree, but not that it's forward in the traversal order.


>That means that every single PhysicalShapeLayer renderered before a ChildSceneLayer in the traversal order will already have the needs_system_composite() flag applied to it, as long as you do the check after PrerollChildren() is called.

That's what we don't want actually. A PhysicalShapeLayer rendered before a ChildSceneLayer should not be system composited.

>I think you could also accomplish this by relying on your flag completely and deleting needs_system_composite(). The 2 flags are IMO redundant with one another, so one of them should go.

I think you're right, but there's a bit of risk there. At the very least it might lead to more calls to UpdateScene than are necessary. If it's ok, i'd prefer to skip entirely if this code goes away soon anyway, or at the very least defer